--- a/vllm/distributed/parallel_state.py	2026-10-02 13:21:30.524197525 +0400
+++ b/vllm/distributed/parallel_state.py	2026-10-02 13:21:30.523246318 +0400
@@ -25,6 +25,7 @@
 
 import contextlib
 import gc
+import os
 import pickle
 import weakref
 from collections import deque, namedtuple
@@ -504,7 +505,9 @@
                 # processes through the CPU.
                 with suppress_stdout():
                     cpu_group = torch.distributed.new_group(
-                        ranks, backend="gloo", timeout=timeout
+                        ranks,
+                        backend=os.environ.get("VLLM_CPU_GROUP_BACKEND", "gloo"),
+                        timeout=timeout,
                     )
                 if self.rank in ranks:
                     self.ranks = ranks
